internal/controller/operator/instance/dapr_instance_controller_action_apply.go

```diff
@@ -23,13 +26,26 @@ import (
 	"github.com/dapr-sandbox/dapr-kubernetes-operator/pkg/resources"
 )
 
-func NewApplyAction() Action {
-	return &ApplyAction{
+const autoPullPolicySidecarInjector = `
+if (.dapr_sidecar_injector.image | has("name")) and (.dapr_sidecar_injector | has("sidecarImagePullPolicy") | not) 
+then 
+  .dapr_sidecar_injector.sidecarImagePullPolicy = "Always"
+end
+`
```

Comment:
@ItalyPaleAle @salaboy 


Here I've added a sort of customizer that automatically set the sidecar pull policy in case of a custom image is set. I'm not sure if having such customization hard-coded in the the operator is a good idea so maybe there should be a ValuesCustomizer CRD that platform admins can use in order to force certain behaviors ?

Reply:
@lburgazzoli this should be configurable.. but let's merge this very long PR first and then we can think of how we customize it.